### PR TITLE
Remove unused imports in generated proxy class

### PR DESCRIPTION
**Pull request by MGaetan89**
It seems that those three imports are never used in the generated proxy class.

### CODE REVIEW

**Comment by dalinaum** [comment]
Thanks for your contribution, @MGaetan89.
LGTM :+1:

**Comment by beeender** [comment]
retest this please

File: realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java [hunk 1]
```diff
@@ -68,17 +68,14 @@ public void generate() throws IOException, UnsupportedOperationException {
         imports.add("io.realm.internal.TableOrView");
         imports.add("io.realm.internal.ImplicitTransaction");
         imports.add("io.realm.internal.LinkView");
-        imports.add("io.realm.internal.android.JsonUtils");
         imports.add("java.io.IOException");
         imports.add("java.util.ArrayList");
         imports.add("java.util.Collections");
         imports.add("java.util.List");
-        imports.add("java.util.Date");
```
Comment (cmelchior) [inline realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: I am pretty sure this is needed if you Model class contains date objects. Will have to look into the others. Unfortunately JavaWriter doesn't allow you to only import the Classes that is actually used :(

Comment (MGaetan89) [inline realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java]: I didn't find any direct usage of the `Date` class. If I missed something, I'll obviously add it back.

**Comment by MGaetan89** [comment]
Any idea why the build fails? I do not have access to the result.

**Comment by emanuelez** [comment]
```
/home/jenkins/jenkins/workspace/java_pr/examples/introExample/build/generated/source/apt/debug/io/realm/PersonRealmProxy.java:296: error: cannot find symbol
                JSONArray array = json.getJSONArray("cats");
                ^
  symbol:   class JSONArray
  location: class PersonRealmProxy
1 error
```

It looks like JSONArray is needed after all. I'm pretty sure the other two are needed as well.

**Comment by cmelchior** [comment]
```
Note: Processing class Cat
Note: Processing class Person
Note: Processing class Dog
Note: Creating DefaultRealmModule
/home/jenkins/jenkins/workspace/java_pr/examples/introExample/build/generated/source/apt/debug/io/realm/PersonRealmProxy.java:296: error: cannot find symbol
                JSONArray array = json.getJSONArray("cats");
                ^
  symbol:   class JSONArray
  location: class PersonRealmProxy
```

Note that the content of a Proxy class depend on the fields  present, e.g. all the above imports might be used by the following class:

```
public class Foo extends RealmObject {
  private Date birthday;
  private RealmList<Foo> foos;
}
```

That is a limitation of JavaWriter really. From my understanding JavaPoet is able to only include the relevant imports and we also wants to migrate at some point, but it hasn't really been pressing so far.

**Comment by cmelchior** [comment]
Hi @MGaetan89 
Thank you for this, but I just went through all our dependencies and all 3 imports are used in some corner cases. Ideally we should dynamically detect when that is, but as long as we use JavaWriter and not JavaPoet that will not be possible.

So for now we will have to close this PR.

**Pull request closed without merging**